### ISSUE TITLE
Error when trying to upscale an image from SDXL model and mix it with an other model (Realistic Vision)

### ISSUE THREAD

**Issue by MattRM2** [ssitu/ComfyUI_UltimateSDUpscale#12]
Hi,

I got this error when I tried to use the Realistic Vision v5.0 model (same with other SD 1.5 model). What I can do to avoid this error ? I think it's about the aspect ratio of the source image and the SD 1.5 model lke Realistic Vision, no ? Or it's something else ;)

I try to use this model during upscale or Photon v1.0 to get more realistic skin / face during upscale. It's working well with standard SD 1.5 model but here, it's not working (note: work well if I use the SDXL refiner model but the skin and the face are not so well, not like a photo).

Here is the error : 
`
Error occurred when executing UltimateSDUpscale:

mat1 and mat2 shapes cannot be multiplied (154x1280 and 768x320)

File "E:\00_ComfyUI\ComfyUI\execution.py", line 144, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
File "E:\00_ComfyUI\ComfyUI\execution.py", line 74, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
File "E:\00_ComfyUI\ComfyUI\execution.py", line 67, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
File "E:\00_ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\nodes.py", line 124, in upscale
processed = script.run(p=sdprocessing, _=None, tile_width=tile_width, tile_height=tile_height,
File "E:\00_ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories/ultimate_sd_upscale/scripts/ultimate-upscale.py", line 553, in run
upscaler.process()
File "E:\00_ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories/ultimate_sd_upscale/scripts/ultimate-upscale.py", line 136, in process
self.image = self.redraw.start(self.p, self.image, self.rows, self.cols)
File "E:\00_ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories/ultimate_sd_upscale/scripts/ultimate-upscale.py", line 245, in start
return self.chess_process(p, image, rows, cols)
File "E:\00_ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories/ultimate_sd_upscale/scripts/ultimate-upscale.py", line 215, in chess_process
processed = processing.process_images(p)
File "E:\00_ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\modules\processing.py", line 119, in process_images
(samples,) = common_ksampler(p.model, p.seed, p.steps, p.cfg, p.sampler_name,
File "E:\00_ComfyUI\ComfyUI\nodes.py", line 1127, in common_ksampler
samples = comfy.sample.sample(model, noise, steps, cfg, sampler_name, scheduler, positive, negative, latent_image,
File "E:\00_ComfyUI\ComfyUI\comfy\sample.py", line 88, in sample
samples = sampler.sample(noise, positive_copy, negative_copy, cfg=cfg, latent_image=latent_image, start_step=start_step, last_step=last_step, force_full_denoise=force_full_denoise, denoise_mask=noise_mask, sigmas=sigmas, callback=callback, disable_pbar=disable_pbar, seed=seed)
File "E:\00_ComfyUI\ComfyUI\comfy\samplers.py", line 714, in sample
samples = k_diffusion_sampling.sample_dpm_adaptive(self.model_k, noise, sigma_min, sigmas[0], extra_args=extra_args, callback=k_callback, disable=disable_pbar)
File "E:\00_ComfyUI\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "E:\00_ComfyUI\ComfyUI\comfy\k_diffusion\sampling.py", line 487, in sample_dpm_adaptive
x, info = dpm_solver.dpm_solver_adaptive(x, dpm_solver.t(torch.tensor(sigma_max)), dpm_solver.t(torch.tensor(sigma_min)), order, rtol, atol, h_init, pcoeff, icoeff, dcoeff, accept_safety, eta, s_noise, noise_sampler)
File "E:\00_ComfyUI\ComfyUI\comfy\k_diffusion\sampling.py", line 438, in dpm_solver_adaptive
eps, eps_cache = self.eps(eps_cache, 'eps', x, s)
File "E:\00_ComfyUI\ComfyUI\comfy\k_diffusion\sampling.py", line 338, in eps
eps = (x - self.model(x, sigma, *args, **self.extra_args, **kwargs)) / self.sigma(t)
File "E:\00_ComfyUI\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\00_ComfyUI\ComfyUI\comfy\samplers.py", line 319, in forward
out = self.inner_model(x, sigma, cond=cond, uncond=uncond, cond_scale=cond_scale, cond_concat=cond_concat, model_options=model_options, seed=seed)
File "E:\00_ComfyUI\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\00_ComfyUI\ComfyUI\comfy\k_diffusion\external.py", line 125, in forward
eps = self.get_eps(input * c_in, self.sigma_to_t(sigma), **kwargs)
File "E:\00_ComfyUI\ComfyUI\comfy\k_diffusion\external.py", line 151, in get_eps
return self.inner_model.apply_model(*args, **kwargs)
File "E:\00_ComfyUI\ComfyUI\comfy\samplers.py", line 307, in apply_model
out = sampling_function(self.inner_model.apply_model, x, timestep, uncond, cond, cond_scale, cond_concat, model_options=model_options, seed=seed)
File "E:\00_ComfyUI\ComfyUI\comfy\samplers.py", line 285, in sampling_function
cond, uncond = calc_cond_uncond_batch(model_function, cond, uncond, x, timestep, max_total_area, cond_concat, model_options)
File "E:\00_ComfyUI\ComfyUI\comfy\samplers.py", line 262, in calc_cond_uncond_batch
output = model_function(input_x, timestep_, **c).chunk(batch_chunks)
File "E:\00_ComfyUI\ComfyUI\comfy\model_base.py", line 61, in apply_model
return self.diffusion_model(xc, t, context=context, y=c_adm, control=control, transformer_options=transformer_options).float()
File "E:\00_ComfyUI\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\00_ComfyUI\ComfyUI\comfy\ldm\modules\diffusionmodules\openaimodel.py", line 834, in forward
h = forward_timestep_embed(module, h, emb, context, transformer_options)
File "E:\00_ComfyUI\ComfyUI\comfy\ldm\modules\diffusionmodules\openaimodel.py", line 97, in forward_timestep_embed
x = layer(x, context, transformer_options)
File "E:\00_ComfyUI\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\00_ComfyUI\ComfyUI\comfy\ldm\modules\attention.py", line 695, in forward
x = block(x, context=context[i], transformer_options=transformer_options)
File "E:\00_ComfyUI\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\00_ComfyUI\ComfyUI\comfy\ldm\modules\attention.py", line 527, in forward
return checkpoint(self._forward, (x, context, transformer_options), self.parameters(), self.checkpoint)
File "E:\00_ComfyUI\ComfyUI\comfy\ldm\modules\diffusionmodules\util.py", line 123, in checkpoint
return func(*inputs)
File "E:\00_ComfyUI\ComfyUI\comfy\ldm\modules\attention.py", line 627, in _forward
n = self.attn2(n, context=context_attn2, value=value_attn2)
File "E:\00_ComfyUI\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\00_ComfyUI\ComfyUI\comfy\ldm\modules\attention.py", line 422, in forward
k = self.to_k(context)
File "E:\00_ComfyUI\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\00_ComfyUI\ComfyUI\comfy\ops.py", line 18, in forward
return torch.nn.functional.linear(input, self.weight, self.bias)
`

If you can tell me what I need to do to solve this, it'll be great.
Thank you,

Matt

**Comment by ssitu** [comment]
Does your model work when using a KSampler instead of this node?

**Comment by MattRM2** [comment]
> Does your model work when using a KSampler instead of this node?

Yes, it works very well (I tried 5 differents models), including with the Ultimate SD upscaler. It's only if I mix an image from the SDXL and a SD 1.5 model in the Ultimate SD Upscale.

Sorry for my long delay to answer you,

Matt

**Comment by ssitu** [comment]
Strange, are you able to share the workflow?

**Comment by MattRM2** [comment]
Hi,

So, I retried it to prepare a graph for you and just before doing it, I updated ComfyUI and all the custom node including yours and now it's working. And the great news it's that the preprocessors for controlnet are working too (long bug, working now just because ComfyUI update).

So maybe the bug was from others dependencies.

I give you my graph in attachment if you want to try it from your side but right now it's working. Just have the Roop from comfyui-reactor-node still have a bug when loading but not your dev, so not your problem ;)

Thank you for your tools,

Matt
[Debug_upscale.zip](https://github.com/ssitu/ComfyUI_UltimateSDUpscale/files/12263355/Debug_upscale.zip)
